default/AI/DiplomaticCorp.py

```diff
@@ -0,0 +1,78 @@
+"""Handle diplomatic messages and response determination."""
+
+import random
+
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+import FreeOrionAI as foAI
+import AIstate
+from freeorion_tools import UserString, UserStringList, chat_on_error, print_error
+
+
+@chat_on_error
+def handle_diplomatic_message(message):  # pylint: disable=invalid-name
+    """Handle a diplomatic message update from the server,
+    such as if another player declares war, accepts peace, or cancels a proposed peace treaty.
+    :param message:
+    :return:
+    """
+    print "Received diplomatic %s message from empire %s to empire %s" % (message.type, message.sender, message.recipient)
+    print "my empire id: %s" % fo.empireID()
+    if message.recipient != fo.empireID():
```

Comment:
I forget if I actually ran into trouble without this, or if it was just preexisting and I figured I'd keep it as 'better safe than sorry'.